packages/hotspot-utils/src/hotspotUtils.ts

```diff
@@ -0,0 +1,162 @@
+import {
+  AccountMeta,
+  Connection,
+  PublicKey,
+  TransactionInstruction,
+  TransactionMessage,
+  VersionedMessage,
+  VersionedTransaction,
+} from "@solana/web3.js";
+import { AnchorProvider, Program, Wallet } from "@project-serum/anchor";
+import { HeliumEntityManager } from "@helium/idls/lib/types/helium_entity_manager";
+import {
+  rewardableEntityConfigKey,
+  iotInfoKey,
+  mobileInfoKey,
+  init,
+} from "@helium/helium-entity-manager-sdk";
+import { subDaoKey } from "@helium/helium-sub-daos-sdk";
+import { Asset, AssetProof, getAssetProof } from "./mplAssetAPI";
+import {
+  PROGRAM_ID as BUBBLEGUM_PROGRAM_ID,
+  createTransferInstruction,
+} from "@metaplex-foundation/mpl-bubblegum";
+import {
+  ConcurrentMerkleTreeAccount,
+  SPL_ACCOUNT_COMPRESSION_PROGRAM_ID,
+  SPL_NOOP_PROGRAM_ID,
+} from "@solana/spl-account-compression";
+
+export type SolHotspot = Awaited<ReturnType<typeof getSolHotspotInfo>>;
+
+export const createHeliumEntityManagerProgram = async ({
```

Comment:
This is dangerous. The program will not be able to `rpc` transactions because `Wallet` cannot sign.

Reply:
@ChewingGlass All I really want to do is determine what the fees are and who's paying them when a hotspot is being updated. I know I can pull the `feePayer` from the `update-meta` transaction, but what I'm not sure about is how to determine the staking fees (e.g. If a hotspot the location doesn't change, there's no staking fee.) Can that be pulled from the transaction as well?

Reply:
or would it be better to pull the asset from the mpl asset api and compare the new location to the previous?

Reply:
I removed this. Just going to simulate the transaction to determine fees.